skimage/feature/peak.py

```diff
@@ -159,3 +159,73 @@ def peak_local_max(image, min_distance=10, threshold_abs=0, threshold_rel=0.1,
         nd_indices = tuple(coordinates.T)
         out[nd_indices] = True
         return out
+
+
+def get_scale_local_maximas(cube_coordinates, laplacian_cube):
```

Comment:
Cython can be tricky to fully optimize, but is usually worth the effort for massive memory savings. If you wanted to share that via a Gist, we could see if the potential exists for further efficiency gains.